File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public TreeNode find(TreeNode current, int val) {
        if (current.val == val) {
            return current;
        }
        if (!Objects.equals(current.right, null) && current.val < val) {
            return find(current.right, val);
        }
        if (!Objects.equals(current.left, null) && current.val > val) {
            return find(current.left, val);
        }
        
        return null;
    }
    
    public TreeNode searchBST(TreeNode root, int val) {
        return find(root, val);
    }
}

File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    
    fun find(current: TreeNode?, `val`: Int): TreeNode? {
        if (current?.`val` == `val`) {
            return current
        }
        if (current?.right != null && current?.`val` < `val`) {
            return find(current.right, `val`)
        }
        if (current?.left != null && current?.`val` > `val`) {
            return find(current.left, `val`)
        }
        return null
    }
    
    fun searchBST(root: TreeNode?, `val`: Int): TreeNode? {
        return find(root, `val`)
    }
}